### ISSUE TITLE
[BUG] Metrodroid can't read CharlieCard properly on OnePlus One & OnePlus 3T

### ISSUE THREAD

**Issue by taivlam** [metrodroid/metrodroid#752]
## Description of the issue

Metrodroid can't seem scan the CharlieCard properly on the OnePlus One and OnePlus 3T, both running LineageOS 18.1 for microG.  I have described how Metrodroid behaves for each device below, since the outcomes are slightly different, as of May 2021.

(Both devices used to work fine back in October-November of 2019 with LOS 16.0 and Metrodroid v2.9.43 and v3.0.0.)

I am not sure if this is an issue with Metrodroid or there is something systematically wrong with NFC on the LOS builds.

## For card-specific issues

- Name of the card type: CharlieCard
- Name of the transit agency: MBTA

> Please export a card dump from Metrodroid.

I wish I could, but I couldn't get Metrodroid to that point.

## Steps to reproduce

1.  Turn on NFC (via Quick Tiles).
2.  Open Metrodroid.
3.  Scan CharlieCard.

## Expected behaviour

For both devices, Metrodroid is supposed to work normally (with the encryption keys I have obtained before via MIFARE Classic Tool).

To contrast, Metrodroid works with the CharlieCard just fine on the OnePlus 5T.

### OnePlus One

Whenever I scan a CharlieCard, either Metrodroid does nothing or it seemingly detects the card, only to say that the NFC service died:
<img src='https://user-images.githubusercontent.com/47955724/118931981-11542880-b937-11eb-8619-3d087202215b.png' width=300/>

Additionally, the NFC status (via the Quick Tile) seems like it can't stay on and "flickers" off, even though I have activated NFC for the OnePlus One.

### OnePlus 3T

Whenever I scan a CharlieCard, Metrodroid scans the CharlieCard but says the card is not supported:
<img src='https://user-images.githubusercontent.com/47955724/118932594-bc64e200-b937-11eb-8e3b-7db95b83f73b.jpeg' width=300/>

The NFC status seems to be consistently turned on for the OnePlus 3T.

## Workarounds (if any)

None - not that I could find.  (Except using a OnePlus 5T.)

## Device Logs

I will try my best to do so and have the device log at hand, if needed.

## Details from About screen:

- Metrodroid version: v3.0.0
- Device Model: OnePlus One (`bacon`), OnePlus 3T (`oneplus3`)
- (Android only) Manufacturer: OnePlus
- Android/iOS version: LineageOS 18.1 for microG (Android 11)
- (Android only) NFC: Yes(?)
- (Android only) MIFARE Classic: Yes

## Checklist

- [✓] The card appears in the "supported cards" list in Metrodroid on my device.
- [✓] I have copied all the details from the about screen into this bug.
- [N/A - Metrodroid won't start reading] For card reading issues, I have exported a card dump from Metrodroid, and can email it privately if needed.
- [✓] I have generated an Android bug report ZIP, and can email it privately if needed.

**Comment by taivlam** [comment]
Yes, I tried the MIFARE Classic Tool (MCT) app, which also used to work on the OnePlus One and OnePlus 3T (at some point), but the MCT app also doesn't work on both devices.

In fact, I've mentioned in an [issue](https://github.com/ikarus23/MifareClassicTool/issues/347) for the MCT app (which I've sort of neglected but now I've returned to it) to see what can be done about this.

**Comment by taivlam** [comment]
Since the related issue in ikarus23/MifareClassicTool#347 was closed, due to upstream LOS reasons, I also close this issue here.

TL;DR: You should probably use a newer Android device with current OEM support and not use the OPO or OP3T anymore in 2023 (and beyond), if you want to avoid practical edge cases (such as this) where you start seeing the disadvantages of using Android devices that have been EOL by the OEM for some time.
